smsapi/Api/Action/Contacts/ContactAdd.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace SMSApi\Api\Action\Contacts;
+
+use DateTime;
+use InvalidArgumentException;
+use SMSApi\Api\Response\Contacts\ContactResponse;
+use SMSApi\Client;
+use SMSApi\Proxy\Proxy;
+
+final class ContactAdd extends ContactsAction
+{
+    private static $PARAMETER_METHODS = [
+        ContactResponse::FIELD_FIRST_NAME => 'setFirstName',
+        ContactResponse::FIELD_LAST_NAME => 'setLastName',
+        ContactResponse::FIELD_PHONE_NUMBER => 'setPhoneNumber',
+        ContactResponse::FIELD_EMAIL => 'setEmail',
+    ];
+
+    public static function fromFirstName($firstName, Client $client, Proxy $proxy)
+    {
+        return new self(ContactResponse::FIELD_FIRST_NAME, $firstName, $client, $proxy);
+    }
+
+    public static function fromLastName($lastName, Client $client, Proxy $proxy)
+    {
+        return new self(ContactResponse::FIELD_LAST_NAME, $lastName, $client, $proxy);
+    }
+
+    public static function fromPhoneNumber($phoneNumber, Client $client, Proxy $proxy)
+    {
+        return new self(ContactResponse::FIELD_PHONE_NUMBER, $phoneNumber, $client, $proxy);
+    }
+
+    public static function fromEmail($email, Client $client, Proxy $proxy)
+    {
+        return new self(ContactResponse::FIELD_EMAIL, $email, $client, $proxy);
+    }
+
+    /**
+     * @param string $parameterName
+     * @param string|int $parameterValue
+     * @param Client $client
+     * @param Proxy $proxy
+     */
+    public function __construct($parameterName, $parameterValue, Client $client, Proxy $proxy)
```

Comment:
I can't make `__construct` private because of extended classes visibility, so I will make it `@internal`.
These factories are for use instead of `__construct`.